pkg/pipeline/encode/encode_json_test.go

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2022 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package encode
+
+import (
+	"encoding/json"
+	"github.com/netobserv/flowlogs2metrics/pkg/config"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func initNewEncodeJson(t *testing.T) Encoder {
+	newEncode, err := NewEncodeJson()
+	require.Equal(t, err, nil)
+	return newEncode
+}
+
+func TestEncodeJson(t *testing.T) {
+	newEncode := initNewEncodeJson(t)
+	encodeJson := newEncode.(*encodeJson)
+	map1 := config.GenericMap{
+		"varInt":    12,
+		"varString": "testString1",
+		"varBool":   true,
+	}
+	map2 := config.GenericMap{
+		"varString": "testString2",
+		"varInt":    14,
+		"varBool":   false,
+	}
+	map3 := config.GenericMap{}
+	var out []interface{}
+	var in []config.GenericMap
+	out = encodeJson.Encode(in)
+	require.Equal(t, 0, len(out))
+	in = append(in, map1)
+	in = append(in, map2)
+	in = append(in, map3)
+	out = encodeJson.Encode(in)
+	require.Equal(t, len(in), len(out))
+	outputString1, _ := json.Marshal(map1)
+	outputString2, _ := json.Marshal(map2)
+	outputString3, _ := json.Marshal(map3)
+	require.Equal(t, string(outputString1), out[0].(string))
```

Comment:
I think the test should use a predefined string of the expected json and compare the output to it, rather than using `json.Marshal()` as in the tested function
```go
	expected1 := `{"varInt":12,"varBool":true,"varString":"testString1"}`
	actual1 := out[0].(string)
	require.JSONEq(t, expected1, actual1)
```

Reply:
We then have to worry about the way the map is converted into a string and the order of the elements. If we use the Marshal function, then we should get the same order as is done by the program.

Reply:
done